JASP-Engine/JASP/R/regressionlinearbayesian.R

```diff
@@ -15,587 +15,360 @@
+    bas_lm[["priorprobs"]] <- bas_lm[["priorprobs"]] / sum(bas_lm[["priorprobs"]])
+    bas_lm[["priorprobsPredictor"]] <- .basregComputePriorMarginalInclusionProbs(bas_lm)
+    bas_lm[["formula"]] <- formula
+    bas_lm[["weights"]] <- wlsWeights
+    bas_lm[["BFinclusion"]] <- .basregComputeInclusionBF(bas_lm)
+    bas_lm[["namesx"]][-1] <- .unvf(bas_lm[["namesx"]][-1])
+  }
+  
+  result <- list(model = bas_lm, status = status)
+  
+  jaspResults[["stateModel"]]$object <- result
+  
+  return(result)
 }
 
-.calcDataRegTable.basReg <- function(bas_obj, status, options) {
-	# Calculate the data needed for the main table
-	#
-	# Args:
-	#   bas_obj: bas object (including nuisanceTerms entry)
-	#   status: current status of the analysis
-	#   options: a list of user options
-	#
-	# Return:
-	#   list with table footnotes and data rows (ready to insert in 'data' of table)
-
-	if (status$error == TRUE) {
-		return(NULL)
-	}
-
-	nuisanceTerms <- bas_obj$nuisanceTerms # vector of TRUE / FALSE
-	if (length(nuisanceTerms) != length(bas_obj$namesx) - 1) {
-	# if there are categorical predictors
-
-		nuisanceTerms2 <- logical(length(bas_obj$namesx) - 1)
-		names(nuisanceTerms2) <- bas_obj$namesx[-1]
-		for (i in which(nuisanceTerms[nuisanceTerms])) {
-
-			idx <- grep(pattern = names(nuisanceTerms[i],
-				x = names(nuisanceTerms2)[i]), fixed = TRUE)
-			nuisanceTerms2[idx] <- TRUE
-		}
-		nuisanceTerms <- nuisanceTerms2
-
-	} else {
-		names(nuisanceTerms) <- .unvf(names(nuisanceTerms))
-	}
-
-	# default number of models to be shown
-	nRows <- NULL
-	if (options$shownModels == "limited") {
-		nRows <- options$numShownModels
-	}
-
-	if (is.null(nRows) || nRows > length(bas_obj$which)) {
-		nRows <- length(bas_obj$which)
-	}
-
-	# ordered indices based on posterior probabilities of the models
-	models.ordered <- order(bas_obj$postprobs, decreasing = TRUE)
-	if (options$bayesFactorOrder == "nullModelTop") {
-		bestModelIndices <- models.ordered[1:nRows]
-		if (1 %in% bestModelIndices) { # null model has index 1
-			index <- which(bestModelIndices == 1)
-			models.ordered <- c(1, bestModelIndices[-index]) # change position of null model
-		} else {
-			nRows <- nRows + 1 # show null model + best n
-			models.ordered <- c(1, bestModelIndices)
-		}
-	} else { # best model top
-		models.ordered <- models.ordered[1:nRows]
-	}
-	models <- bas_obj$which[models.ordered]
+.basregTableModelComparison <- function(basregModel, jaspResults, options, status) {
+  if (!is.null(jaspResults[["modelComparisonTable"]]))
+    return()
+  
+  modelComparison <- createJaspTable(title = "Model Comparison")
+  jaspResults[["modelComparisonTable"]] <- modelComparison
+  
+  modelComparison$copyDependenciesFromJaspObject(jaspResults[["stateModel"]])
+  modelComparison$dependOnOptions(c("bayesFactorType", "bayesFactorOrder", "shownModels", "numShownModels"))
+  
+  
+  # modelComparison$addCitation <- list(
```

Comment:
It's a bit ugly but you can add citations individually:
```r
modelComparison$addCitation("Clyde, M. A. (2017). BAS: Bayesian Adaptive Sampling for Bayesian Model Averaging. (Version 1.4.7)[Computer software].")
modelComparison$addCitation("Clyde, M. A., Ghosh, J., & Littman, M. L. (2011). Bayesian adaptive sampling for variable selection and model averaging. Journal of Computational and Graphical Statistics, 20, 80-101.")
```

Reply:
Guess this method requires some work too ;)